packages/amplify-graphql-api-construct-tests/src/sql-tests-common/sql-models-auto-increment.ts

```diff
@@ -0,0 +1,187 @@
+
+      expect(getCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(getCoffeeOrder1.customer).toEqual(createCoffeeOrder1.customer);
+
+      // Update Todo Mutation
+      const updateCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.update({
+        orderNumber: createCoffeeOrder1.orderNumber,
+        customer: 'petesv',
+        order: 'hot brew',
+      });
+
+      expect(updateCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(updateCoffeeOrder1.order).toEqual('hot brew');
+
+      // Get Todo Query after update
+      const getUpdatedCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.get({ orderNumber: createCoffeeOrder1.orderNumber });
+
+      expect(getUpdatedCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(getUpdatedCoffeeOrder1.order).toEqual('hot brew');
+
+      // List Todo Query
+      const createCofffeeOrder2 = await coffeeQueueTableCRUDLHelper.create({ order: 'latte' });
+      const listTodo = await coffeeQueueTableCRUDLHelper.list();
+
+      expect(listTodo.items.length).toEqual(2);
+      expect(listTodo.items).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            orderNumber: getUpdatedCoffeeOrder1.orderNumber,
+            order: 'hot brew',
+          }),
+          expect.objectContaining({
+            orderNumber: createCofffeeOrder2.orderNumber,
+            order: 'latte',
+          }),
+        ]),
+      );
+
+      // Delete Todo Mutation
+      const deleteCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.delete({ orderNumber: createCoffeeOrder1.orderNumber });
+
+      expect(deleteCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(deleteCoffeeOrder1.order).toEqual('hot brew');
+
+      const getDeletedCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.get({ orderNumber: createCoffeeOrder1.orderNumber });
+
+      expect(getDeletedCoffeeOrder1).toBeNull();
+
+      // List Todo Query after delete
+      const listCoffeeOrdersAfterDelete = await coffeeQueueTableCRUDLHelper.list();
+
+      expect(listCoffeeOrdersAfterDelete.items.length).toEqual(1);
+      expect(listCoffeeOrdersAfterDelete.items).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            orderNumber: createCofffeeOrder2.orderNumber,
+            order: 'latte',
+          }),
+        ]),
+      );
+
+      // Check invalid CRUD operation returns generic error message
+      const createTodo6 = await coffeeQueueTableCRUDLHelper.create({ order: 'mocha' });
+
+      try {
+        await coffeeQueueTableCRUDLHelper.create({ orderNumber: createTodo6.orderNumber, order: 'americano' });
+      } catch (error) {
+        console.log('throw in create');
+        coffeeQueueTableCRUDLHelper.checkGenericError(error?.message);
+      }
+
+      const invalidOrderNumber = 99999999;
```

Comment:
I assume this is invalid because it exceeds the maximum value of a `SERIAL` field? If so, worth noting that in a comment, in case we start supporting PG versions that have a higher limit.

Reply:
Actually it is invalid because it is greater than any existing record's `orderNumber`. That purpose is definitely not clear since even the number 4 _should_ be invalid. I wrote some comments but then opted to use a more descriptive variable name:
```
const biggerThanAnyExistingOrderNumber = 99999999;
```